cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -389,6 +390,42 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         }
     });
 
+    SelfPacedDueDateEditor = AbstractEditor.extend({
+        fieldName: 'due_num_weeks',
+        templateName: 'self-paced-due-date-editor',
+        className: 'modal-section-content has-actions due-date-input grading-due-date',
+
+        events: {
+            'click .clear-date': 'clearValue'
+        },
+
+        getValue: function() {
+            return parseInt(this.$('#due_in').val());
+        },
+
+        clearValue: function(event) {
+            event.preventDefault();
+            this.$('#due_in').val('');
+        },
+
+        afterRender: function() {
+            AbstractEditor.prototype.afterRender.call(this);
+            this.setDueIn(this.model.get('due_num_weeks'));
+        },
+
+        setDueIn: function(value) {
+            this.$('.field-due-in input').val(value);
+        },
```

Comment:
`DueDateEditor` is [excluded in the code](https://github.com/edx/edx-platform/blob/f39f9d6149a97dfd7ef0bc0c8d5ba801be4b1270/cms/static/js/views/modals/course_outline_modals.js#L1138-L1144) for self paced courses so I wanted another editor that I could add instead of removing this code (and also since I don't need the time input and I am trying to save a different metadatum). But do you think it might better to restrict those conditions within DueDateEditor and get rid of SelfPacedDueDateEditor?

Reply:
Oh sorry, no. I think you're totally good in the way you have it implemented. I meant specifically the `setDueIn` function. I don't see a setter in the `DueDateEditor` so trying to understand why it's necessary here

Reply:
Ahh yeah so `DueDateEditor` seems to be an extension of [`BaseDateEditor` where its `afterRender` function](https://github.com/edx/edx-platform/blob/f39f9d6149a97dfd7ef0bc0c8d5ba801be4b1270/cms/static/js/views/modals/course_outline_modals.js#L346-L368) works only for an editor with both date and time input. Since I only have the number input, I wanted a different afterRender function. But now that I'm looking at the code I think I'm just going to delete `setDueIn` and put that one liner into [`afterRender` function](https://github.com/edx/edx-platform/blob/f39f9d6149a97dfd7ef0bc0c8d5ba801be4b1270/cms/static/js/views/modals/course_outline_modals.js#L411-L414) that calls it!

Reply:
Ah yeah, I see what you mean. I love that thinking! Sounds great